pkg/domain/infra/abi/pods.go

```diff
@@ -83,6 +83,57 @@ func (ic *ContainerEngine) PodKill(ctx context.Context, namesOrIds []string, opt
 	return reports, nil
 }
 
+func (ic *ContainerEngine) PodLogs(ctx context.Context, nameOrID string, options entities.PodLogsOptions) error {
+	// Implementation accepts slice
+	podName := []string{nameOrID}
+	pod, err := getPodsByContext(false, false, podName, ic.Libpod)
+	if err != nil {
+		return err
+	}
+	// Get pod containers
+	podCtrs, err := pod[0].AllContainers()
+	if err != nil {
+		return err
+	}
+
+	ctrNames := []string{}
+	// Check if `kubectl pod logs -c ctrname <podname>` alike command is used
+	if options.ContainerName != "" {
+		ctrFound := false
+		for _, ctr := range podCtrs {
+			if ctr.ID() == options.ContainerName || ctr.Name() == options.ContainerName {
+				ctrNames = append(ctrNames, options.ContainerName)
+				ctrFound = true
+			}
+		}
+		if !ctrFound {
+			return errors.Wrapf(define.ErrNoSuchCtr, "container %s is not in pod %s", options.ContainerName, nameOrID)
+		}
+	} else {
+		// No container name specified select all containers
+		for _, ctr := range podCtrs {
+			ctrNames = append(ctrNames, ctr.Name())
+		}
+	}
+
+	// PodLogsOptions are similar but contains few extra fields like ctrName
+	// So cast other values as is so we can re-use the code
+	containerLogsOpts := entities.ContainerLogsOptions{
+		Details:      options.Details,
```

Comment:
Would adding JSON tags and simply encoding/decoding make more sense here so we don't have to have these big blocks of assignments?

Reply:
@cdoern thanks for catching this :) I agree we can wrap this in a helper function however JSON encoding/decoding seems inconsistent with how we are handling opts in entire codebase. 

Fixed in latest commit.